src/subdomains/monitoring/entities/monitoring-balance.entity.ts

```diff
@@ -16,9 +18,36 @@ export class MonitoringBalanceEntity extends IEntity {
   @Column({ type: 'float' })
   customerBalance: number;
 
+  @Column({ type: 'float', default: 0 })
+  assetCHF: number;
+
+  @Column({ type: 'float', default: 0 })
+  ldsBalance: number;
+
+  @Column({ type: 'float', default: 0 })
+  ldsBalanceCHF: number;
+
   // --- ENTITY METHODS --- //
 
   checksum(): number {
     return this.onchainBalance + this.lightningBalance + this.customerBalance;
   }
+
+  updateBtcBalance(chfPrice: Price): this {
+    this.ldsBalance = this.onchainBalance + this.lightningBalance - this.customerBalance;
+
+    this.assetCHF = chfPrice.convert(1, 2);
+    this.ldsBalanceCHF = chfPrice.convert(LightningHelper.satToBtc(this.ldsBalance), 2);
+
+    return this;
+  }
+
+  updateChfBalance(): this {
+    this.ldsBalance = this.onchainBalance + this.lightningBalance - this.customerBalance;
+
+    this.assetCHF = 1;
+    this.ldsBalanceCHF = this.ldsBalance;
+
+    return this;
+  }

```

Comment:
I would put this into two separate create methods. It's a bit weird to create the entity and then call some methods depending on the type 🤔